rustls/src/verify.rs

```diff
@@ -321,6 +321,64 @@ impl fmt::Debug for dyn ClientCertVerifier {
     }
 }
 
+/// Verify that the end-entity certificate `end_entity` is a valid server cert
+/// and chains to at least one of the [OwnedTrustAnchor] in the `roots` [RootCertStore].
+///
+/// `intermediates` contains all certificates other than `end_entity` that
+/// were sent as part of the server's [Certificate] message. It is in the
+/// same order that the server sent them and may be empty.
+#[allow(unreachable_pub, dead_code)]
```

Comment:
similar suggestion here: I think the `unreachable_pub` allow exception should be specific to `not(feature = "dangerous_configuration")`:

```suggestion
#[allow(dead_code)]
#[cfg_attr(not(feature = "dangerous_configuration"), allow(unreachable_pub))]
```